src/components/subscribe.tsx

```diff
@@ -0,0 +1,142 @@
+    this.setState({ email: e.target.value });
+  };
+
+  handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const email = this.state.email || null;
+    const { i18n } = this.props;
+    const locale = i18n.language;
+
+    // check if email is missing, return undefined
+    if (!email) {
+      this.setState({
+        response: i18n._(t`Please enter an email address!`),
+      });
+      return;
+    }
+
+    const tenantPlatformOrigin =
+      process.env.GATSBY_TENANT_PLATFORM_SITE_ORIGIN ||
+      "https://demo.justfix.nyc";
+
+    fetch(`${tenantPlatformOrigin}/mailchimp/subscribe`, {
+      method: "POST",
+      mode: "cors",
+      body: `email=${encodeURIComponent(
+        email
+      )}&language=${locale}&source=orgsite`,
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+    })
+      .then((result) => result.json())
+      .then((result) => {
+        if (result.status === 200) {
+          this.setState({
+            success: true,
+            response: i18n._(t`All set! Thanks for subscribing!`),
+          });
+        } else if (result.errorCode === "INVALID_EMAIL") {
+          this.setState({
+            response: i18n._(t`Oops! That email is invalid.`),
+          });
+        } else {
+          this.setState({
+            response: i18n._(
+              t`Oops! A network error occurred. Try again later.`
+            ),
+          });
+        }
+      })
+      .catch((err) => {
+        this.setState({
+          response: i18n._(t`Oops! A network error occurred. Try again later.`),
+        });
+      });
+  };
+
+  render() {
+    const { i18n, location } = this.props;
+
+    // Default styling is for "footer"
+    const defaultResponseTextClass =
+      location === "page" ? "has-text-grey-dark" : "has-text-white";
+
+    return (
+      <div>
+        <form
+          className="email-form is-horizontal-center"
+          onSubmit={this.handleSubmit}
+        >
+          <div
+            className={classnames(location === "page" && "field has-addons")}
+          >
+            <div className="control is-expanded">
+              <input
+                type="email"
+                name="EMAIL"
+                className="input"
+                onChange={this.handleChange}
+                placeholder={i18n._(t`Email Address`)}
+              />
+            </div>
+            <div className="control has-text-centered-touch">
+              <button className="button is-primary is-uppercase" type="submit">
+                <Trans>Sign up</Trans>
+              </button>
+            </div>
+          </div>
+        </form>
+        {this.state.response && (
+          <>
+            <p
+              className={
+                this.state.success
+                  ? defaultResponseTextClass
+                  : "has-text-danger"
+              }
+            >
+              {this.state.response}
+            </p>
```

Comment:
Urf, so one thing about this approach (and probably the WoW one at https://github.com/JustFixNYC/who-owns-what/pull/288 too) is that these messages don't have any kind of ARIA support, which means that screen reader users will not hear them.  I think we can address this later, though, since we'll likely have to do it via an ARIA announcer component, which is its own can of worms, so maybe just file an issue about it for now?

Reply:
Sounds good— just made a note of this in #114